### DISCUSSION TITLE
Middleware for controllers

### DISCUSSION THREAD

**Discussion by kierangilliam** [strapi/strapi#6061] [Legacy Topics]
Hello,

I am looking to publish to a Redis PubSub stream every time certain objects are Created, Updated, or Deleted. I know I can do the below code in the controller file for each model that I want this middleware, but it feels like a bit much to copy the [core controllers](https://strapi.io/documentation/3.0.0-beta.x/concepts/controllers.html#core-controllers) into each model just to add the one function `strapi.services.pubsub.publish()`. Does anyone have any ideas for a more elegant way to achieve this?

```
module.exports = {

    async create(ctx) {
        let entity
      
        if (ctx.is('multipart')) {
            const { data, files } = parseMultipartData(ctx)
            entity = await strapi.services.mymodel.create(data, { files })
        } else {
            entity = await strapi.services.mymodel.create(ctx.request.body)
        }

        strapi.services.pubsub.publish(
            entity.id,
            {
                id: entity.id,
                typename: 'MyModel',
                op: 'CREATE',
            }
        )

        ctx.log.info(`Created model: ${entity.id}`) 
        return sanitizeEntity(entity, { model: strapi.models.mymodel })
    },

    async update(ctx) {
        ....
    }

    async delete(ctx) {
        ....
    }
}
```

**Comment by robertpiosik** [comment]
I would use [lifecycle hooks](https://strapi.io/documentation/3.0.0-beta.x/concepts/models.html#concept) for that.

**Reply by malaDev** [reply]
@robertpiosik I guess it's the part in `api/<name>/models/<name>.js`. Is there a way to implement same logic on all models without repeating it? example I have dozen of resources then I want to run a publishing statement on each create/update/delete for all the resources. I checked the doc and I was not able to figure this out.
